// [[Reason: Invalid computed property can only be detected by TSC]] ////

//// [tests/cases/conformance/es6/computedProperties/computedPropertyNamesWithStaticProperty.ts] ////

===================================================================
--- TSC	declarations
+++ DTE	declarations
@@ -4,49 +4,47 @@
 declare class C {
 }
 declare class C1 {
     static staticProp: number;
+    get [C1.staticProp](): string;
+    set [C1.staticProp](x: string);
+    [C1.staticProp](): invalid;
 }
 
 /// [Errors] ////
 
 computedPropertyNamesWithStaticProperty.ts(2,1): error TS1068: Unexpected token. A constructor, method, accessor, or property was expected.
-computedPropertyNamesWithStaticProperty.ts(4,9): error TS9014: Computed properties must be number or string literals, variables or dotted expressions with --isolatedDeclarations.
 computedPropertyNamesWithStaticProperty.ts(4,10): error TS2449: Class 'C1' used before its declaration.
-computedPropertyNamesWithStaticProperty.ts(7,9): error TS9014: Computed properties must be number or string literals, variables or dotted expressions with --isolatedDeclarations.
 computedPropertyNamesWithStaticProperty.ts(7,10): error TS2449: Class 'C1' used before its declaration.
-computedPropertyNamesWithStaticProperty.ts(10,5): error TS9014: Computed properties must be number or string literals, variables or dotted expressions with --isolatedDeclarations.
+computedPropertyNamesWithStaticProperty.ts(10,5): error TS9008: Method must have an explicit return type annotation with --isolatedDeclarations.
 computedPropertyNamesWithStaticProperty.ts(10,6): error TS2449: Class 'C1' used before its declaration.
 computedPropertyNamesWithStaticProperty.ts(15,10): error TS2449: Class 'C2' used before its declaration.
 computedPropertyNamesWithStaticProperty.ts(18,10): error TS2449: Class 'C2' used before its declaration.
 computedPropertyNamesWithStaticProperty.ts(21,6): error TS2449: Class 'C2' used before its declaration.
 
 
-==== computedPropertyNamesWithStaticProperty.ts (10 errors) ====
+==== computedPropertyNamesWithStaticProperty.ts (8 errors) ====
     class C {
     class C1 {
     ~~~~~
 !!! error TS1068: Unexpected token. A constructor, method, accessor, or property was expected.
         static staticProp = 10;
         get [C1.staticProp]() {
-            ~~~~~~~~~~~~~~~
-!!! error TS9014: Computed properties must be number or string literals, variables or dotted expressions with --isolatedDeclarations.
              ~~
 !!! error TS2449: Class 'C1' used before its declaration.
 !!! related TS2728 computedPropertyNamesWithStaticProperty.ts:2:7: 'C1' is declared here.
             return "hello";
         }
         set [C1.staticProp](x: string) {
-            ~~~~~~~~~~~~~~~
-!!! error TS9014: Computed properties must be number or string literals, variables or dotted expressions with --isolatedDeclarations.
              ~~
 !!! error TS2449: Class 'C1' used before its declaration.
 !!! related TS2728 computedPropertyNamesWithStaticProperty.ts:2:7: 'C1' is declared here.
             var y = x;
         }
         [C1.staticProp]() { }
         ~~~~~~~~~~~~~~~
-!!! error TS9014: Computed properties must be number or string literals, variables or dotted expressions with --isolatedDeclarations.
+!!! error TS9008: Method must have an explicit return type annotation with --isolatedDeclarations.
+!!! related TS9034 computedPropertyNamesWithStaticProperty.ts:10:5: Add a return type to the method
          ~~
 !!! error TS2449: Class 'C1' used before its declaration.
 !!! related TS2728 computedPropertyNamesWithStaticProperty.ts:2:7: 'C1' is declared here.
     }
